### PR TITLE
LADX: fix modifying item pool in pre_fill

### PR DESCRIPTION
**Pull request by zig-for**
## What is this fixing or adding?
The enforced start item was happening a bit too early, this moves it into create_items

## How was this tested?
Genned a few times, ran the test in the other branch

### CODE REVIEW

File: worlds/ladx/__init__.py [hunk 1]
```diff
@@ -294,6 +299,7 @@ def force_start_item(self):
                 index = self.multiworld.random.choice(possible_start_items)
                 start_item = self.multiworld.itempool.pop(index)
                 start_loc.place_locked_item(start_item)
+                self.pre_fill_items.append(start_item)
```
[the remark this answers is not in the document]
Comment (zig-for) [reply worlds/ladx/__init__.py:302]: I'm really not sure!

[the remark this answers is not in the document]
Comment (zig-for) [reply worlds/ladx/__init__.py]: Fixed. Do I need _any_ of my uses of that function?

**Comment by BootsinSoots** [comment]
Sounds like this is related to #1460?

**Review by black-sliver — COMMENTED** [review]
I believe this will break plando on Tarin's Gift (Mabe Village). Is this expected?

(Especially if we move it to create_items,) `enumerate(self.multiworld.itempool)` should be replaced by a local list of candidates, because iterating over all of multiworld is kinda slow and gets slower with each added slot.

**Comment by zig-for** [comment]
In my ER branch I've rewritten to do exactly that -
https://github.com/ArchipelagoMW/Archipelago/pull/1613/files#diff-3810d5b98b98412cb09c53bf2d31f49f7c56de2993b988e4471fd9cbf32dfaecR520

**Comment by ThePhar** [comment]
> In my ER branch I've rewritten to do exactly that - [#1613 (files)](https://github.com/ArchipelagoMW/Archipelago/pull/1613/files#diff-3810d5b98b98412cb09c53bf2d31f49f7c56de2993b988e4471fd9cbf32dfaecR520)

Does this PR need to wait on #1613 then?

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**